sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy

```diff
@@ -96,222 +94,345 @@ class APISpec extends Specification {
 
-    /*
-    credential for various kinds of accounts.
-     */
-    @Shared
-    static SharedKeyCredential primaryCreds
+    static def AZURE_TEST_MODE = "AZURE_TEST_MODE"
+    static def PRIMARY_STORAGE = "PRIMARY_STORAGE_"
+    static def SECONDARY_STORAGE = "SECONDARY_STORAGE_"
+    static def BLOB_STORAGE = "BLOB_STORAGE_"
+    static def PREMIUM_STORAGE = "PREMIUM_STORAGE_"
 
-    @Shared
-    static SharedKeyCredential alternateCreds
+    static SharedKeyCredential primaryCredential
+    static SharedKeyCredential alternateCredential
+    static SharedKeyCredential blobCredential
+    static SharedKeyCredential premiumCredential
+    static TestMode testMode
 
-    /*
-    URLs to various kinds of accounts.
-     */
-    BlobServiceClient primaryServiceURL
+    BlobServiceClient primaryServiceClient
+    BlobServiceClient alternateServiceClient
+    BlobServiceClient blobStorageServiceClient
+    BlobServiceClient premiumServiceClient
 
-    @Shared
-    static BlobServiceClient alternateServiceURL
+    private InterceptorManager interceptorManager
+    private TestResourceNamer resourceNamer
+    private String testName
 
-    @Shared
-    static BlobServiceClient blobStorageServiceURL
+    def setupSpec() {
+        testMode = setupTestMode()
+        primaryCredential = getCredential(PRIMARY_STORAGE)
+        alternateCredential = getCredential(SECONDARY_STORAGE)
+        blobCredential = getCredential(BLOB_STORAGE)
+        premiumCredential = getCredential(PREMIUM_STORAGE)
+    }
 
-    @Shared
-    static BlobServiceClient premiumServiceURL
+    def setup() {
+        String fullTestName = specificationContext.getCurrentIteration().getName().replace(' ', '').toLowerCase()
+        String className = specificationContext.getCurrentSpec().getName()
 
-    /*
-    Constants for testing that the context parameter is properly passed to the pipeline.
-     */
-    static final String defaultContextKey = "Key"
+        int iterationIndex = fullTestName.lastIndexOf("[")
+        int substringIndex = (int) Math.min((iterationIndex != -1) ? iterationIndex : fullTestName.length(), 50)
+        this.testName = fullTestName.substring(0, substringIndex)
+        this.interceptorManager = new InterceptorManager(className + fullTestName, testMode)
+        this.resourceNamer = new TestResourceNamer(className + testName, testMode, interceptorManager.getRecordedData())
 
-    static String getTestName(ISpecificationContext ctx) {
-        return ctx.getCurrentFeature().name.replace(' ', '').toLowerCase()
-    }
+        primaryServiceClient = setClient(primaryCredential)
+        alternateServiceClient = setClient(alternateCredential)
+        blobStorageServiceClient = setClient(blobCredential)
+        premiumServiceClient = setClient(premiumCredential)
 
-    def generateContainerName() {
-        generateContainerName(specificationContext, iterationNo, entityNo++)
+        cu = primaryServiceClient.getContainerClient(generateContainerName())
```

Comment:
cu -> containerClient

Reply:
I think James is changing this in another PR and I don't want to cause a ton of tiny merge conflicts.